spec/requests/api/v0.1/put_trainee_spec.rb

```diff
@@ -190,6 +191,70 @@
         end
       end
     end
+
+    it "returns status 422 with an invalid `itt_start_date` and the trainee is not updated" do
+      original_itt_start_date = trainee.itt_start_date
+      put(
+        "/api/v0.1/trainees/#{trainee.slug}",
+        headers: { Authorization: "Bearer #{token}" },
+        params: { data: { itt_start_date: "2023-02-30" } },
+      )
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(response.parsed_body).to have_key("errors")
```

Comment:
Could you test the errors similar to [spec/requests/api/v0.1/post_hesa_trainees_spec.rb:149](https://github.com/DFE-Digital/register-trainee-teachers/pull/4273/files#diff-ac6f5417f01e7647c6deda8562e1a72b14b4ea5b079ce21ad9d1c4af53211b46R149)